java/src/main/java/com/azure/graph/bulk/impl/GremlinDocumentOperationCreator.java

```diff
@@ -0,0 +1,153 @@
+     *
+     * @param vertices Stream of objects that are either GremlinVertex objects, or domain objects with the
+     *                 GremlinVertex annotations
+     * @return Stream of Cosmos Item Operations
+     */
+    public Stream<CosmosItemOperation> getVertexUpsertOperations(Stream<Object> vertices) {
+        return vertices.map(v -> {
+            GremlinVertex vertex = (v instanceof GremlinVertex) ?
+                    (GremlinVertex) v :
+                    ObjectToVertex.toGremlinVertex(v);
+            vertex.validate();
+            return getVertexUpsertOperation(vertex);
+        });
+    }
+
+    private CosmosItemOperation getVertexCreateOperation(GremlinVertex vertex) {
+        PartitionKey partitionKey = new PartitionKey(vertex.getPartitionKey().getValue());
+        try {
+            return CosmosBulkOperations.getCreateItemOperation(
+                    new JsonSerializable(mapper.writeValueAsString(vertex)),
+                    partitionKey);
+        } catch (JsonProcessingException e) {
+            throw new DocumentSerializationException(e);
+        }
+    }
+
+    private CosmosItemOperation getVertexUpsertOperation(GremlinVertex vertex) {
+        PartitionKey partitionKey = new PartitionKey(vertex.getPartitionKey().getValue());
+        try {
+            return CosmosBulkOperations.getUpsertItemOperation(
+                    new JsonSerializable(mapper.writeValueAsString(vertex)),
+                    partitionKey);
+        } catch (JsonProcessingException e) {
+            throw new DocumentSerializationException(e);
+        }
+    }
+
+    /**
+     * Used to convert the stream of objects provided into a stream of Cosmos Create Operations
+     *
+     * @param edges Stream of objects that are either GremlinEdge objects, or domain objects with the
+     *              GremlinEdge annotations
+     * @return Stream of Cosmos Item Operations
+     */
+    public Stream<CosmosItemOperation> getEdgeCreateOperations(Stream<Object> edges) {
+        return edges.map(e -> {
+            GremlinEdge edge = e instanceof GremlinEdge ? (GremlinEdge) e : ObjectToEdge.toGremlinEdge(e);
+            edge.validate();
+            return getEdgeCreateOperation(edge);
+        });
+    }
+
+    /**
+     * Used to convert the stream of objects provided into a stream of Cosmos Upsert Operations
+     *
+     * @param edges Stream of objects that are either GremlinEdge objects, or domain objects with the
+     *              GremlinEdge annotations
+     * @return Stream of Cosmos Item Operations
+     */
+    public Stream<CosmosItemOperation> getEdgeUpsertOperations(Stream<Object> edges) {
+        return edges.map(e -> {
+            GremlinEdge edge = e instanceof GremlinEdge ? (GremlinEdge) e : ObjectToEdge.toGremlinEdge(e);
+            edge.validate();
+            return getEdgeUpsertOperation(edge);
+        });
+    }
+
+    private CosmosItemOperation getEdgeCreateOperation(GremlinEdge edge) {
```

Comment:
I think it would be good if this is public so that customers can use single op creation vs creating a Stream if they don't need it.